var/spack/repos/builtin/packages/r-genelendatabase/package.py

```diff
@@ -0,0 +1,39 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class RGenelendatabase(RPackage):
+    """Length of mRNA transcripts for a number of genomes and gene ID
+       formats, largely based on UCSC table browser"""
+
+    homepage = "https://bioconductor.org/packages/release/data/experiment/html/geneLenDataBase.html"
+    git      = "https://git.bioconductor.org/packages/geneLenDataBase.git"
+
+    version('1.16.0', commit='c2a8b2359c6c59388853d6f6d15d71dffb17a198')
+
+    depends_on('r@3.5.0:3.5.9', when='@1.16.0:')
+    depends_on('r-rtracklayer', type='run')
+    depends_on('r-genomicfeatures@1.3.15:', type='run')
```

Comment:
r-tracklayer & r-genomicfeatures are listed as imports, but not dependencies which means they're just needed at runtime.

For r itself, I wasn't sure if I needed to add the type since it is an Rpackage ?

Reply:
I don't know much about R, but going by http://spack.readthedocs.io/en/latest/build_systems/rpackage.html#r-dependencies I would make them all `type=('build', 'run')`.

As for R itself, technically there's a bug where `extends` always adds `type=('build', 'link')`, and `RPackage` also adds `type=('build', 'run')`, so they're effectively `type=('build', 'link', 'run')`, but I plan on fixing this someday so `type=('build', 'run')` is more clear and future-proof.